controllers/external/util.go

```diff
@@ -28,21 +28,21 @@ import (
 )
 
 // Get uses the client and reference to get an external, unstructured object.
-func Get(c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+func Get(ctx context.Context, c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
 	obj := new(unstructured.Unstructured)
 	obj.SetAPIVersion(ref.APIVersion)
 	obj.SetKind(ref.Kind)
 	obj.SetName(ref.Name)
 	key := client.ObjectKey{Name: obj.GetName(), Namespace: namespace}
-	if err := c.Get(context.Background(), key, obj); err != nil {
+	if err := c.Get(ctx, key, obj); err != nil {
 		return nil, err
 	}
 	return obj, nil
 }
 
 // CloneTemplate uses the client and the reference to create a new object from the template.
 func CloneTemplate(c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
```

Comment:
Should `CloneTemplate` also be updated to take a context as well?

Reply:
`CloneTemplate` is called from `syncReplicas` functions. So maybe `syncReplicas` should be called with context as well? 
https://github.com/kubernetes-sigs/cluster-api/blob/d3e2e807ea7611c9f5b1fb8aac995995f26a8484/controllers/machineset_controller.go#L246

Reply:
Likely, I think we can handle each of these cases iteratively rather than trying to change everything at once. Let's create followup issues and go from there.

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/1557